core/consensus/babe/impl/babe_impl.cpp

```diff
@@ -378,6 +379,24 @@ namespace kagome::consensus {
 
     auto block = pre_seal_block_res.value();
 
+    // Ensure block's extrinsics root matches extrinsics in block's body
+    BOOST_ASSERT_MSG(
+        [&block] {
+          std::vector<common::Buffer> buf_exts{};
+          std::transform(block.body.begin(),
+                         block.body.end(),
+                         std::back_inserter(buf_exts),
+                         [](const auto &ext) {
+                           return common::Buffer{scale::encode(ext).value()};
+                         });
+
+          const auto &ext_root_res = storage::trie::calculateOrderedTrieHash(
+              buf_exts.begin(), buf_exts.end());
+          return ext_root_res.has_value() and (ext_root_res.value()
+                        == common::Buffer(block.header.extrinsics_root));
```

Comment:
It seems we don't need additional ```vector``` to keep temporary encoded values. We can extend ```calculateOrderedTrieHash``` with transform functor:
```c++
const auto &ext_root_res = storage::trie::calculateOrderedTrieHash(block.body.begin(), block.body.end(), [](const auto &ext){ 
   return common::Buffer{scale::encode(ext).value()};
});